libs/MeshKernel/src/Mesh2D.cpp

```diff
@@ -676,31 +676,33 @@ void Mesh2D::InitialiseBoundaryNodeClassification()
             continue;
         }
 
-        if (m_nodesTypes[firstNode] == -1 || m_nodesTypes[secondNode] == -1)
+        if (intNodeType[firstNode] == -1 || intNodeType[secondNode] == -1)
         {
             continue;
         }
 
         if (m_edgesNumFaces[e] == 0)
         {
-            m_nodesTypes[firstNode] = -1;
-            m_nodesTypes[secondNode] = -1;
+            intNodeType[firstNode] = -1;
```

Comment:
i do not like this logic with the node type, but I do not think we can do much about it

Reply:
and its good to keep the same integer values as dflowfm, making it easier to debug